projects/aca-shared/src/lib/components/open-in-app/open-in-app.component.ts

```diff
@@ -0,0 +1,49 @@
+/*!
+ * @license
+ * Alfresco Example Content Application
+ *
+ * Copyright (C) 2005 - 2020 Alfresco Software Limited
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component, Inject, ViewEncapsulation } from '@angular/core';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+@Component({
+  selector: 'aca-open-in-app',
+  templateUrl: './open-in-app.component.html',
+  styleUrls: ['./open-in-app.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class OpenInAppComponent {
+  private redirectUrl: string;
+  public window = window;
+  constructor(
+    @Inject(MAT_DIALOG_DATA)
+    public data: any
+  ) {
+    if (data) {
+      this.redirectUrl = data.redirectUrl;
+    }
+  }
+  openInApp() {
```

Comment:
While I agree it's best to be explicit with the return types and they absolutely should be annotated where possible, adding it for a `void` type seems to me a little redundant as the lack of a return statement makes this clear already. Just my opinion.

Reply:
As far as I saw there is a plan to add it as lint rule -> https://alfresco.atlassian.net/browse/AAE-11899 from this conversation -> https://hyland.slack.com/archives/C016SMNNL8L/p1672328103754219 so probably it would be best to get used to adding this. cc @DenysVuika should we also follow and this lint rule?

Reply:
Interesting, thank you for bringing that to my attention. If that's the agreed upon code style and we're enabling the lint rule for it then I have no real opposition to it. Works either way for me.